connectors/src/connectors/zendesk/temporal/workflows.ts

```diff
@@ -388,6 +394,50 @@ export async function zendeskCategorySyncWorkflow({
   }
 }
 
+/**
+ * Garbage collection workflow for Zendesk.
+ *
+ * This workflow is responsible for deleting the following (in this order):
+ * - Outdated tickets.
+ * - Articles that cannot be found anymore in the Zendesk API.
+ * - Categories that have no article anymore.
+ * - Brands that have no permission on tickets and Help Center anymore.
+ */
+export async function zendeskGarbageCollectionWorkflow({
+  connectorId,
+}: {
+  connectorId: ModelId;
+}) {
+  // deleting the outdated tickets
+  let ticketIds = await getNextOldTicketBatchActivity(connectorId);
```

Comment:
nit (feel free to not address this, can be a future refactoring one day if needed)

We could do `deleteOldTicketBatchActivity(..) -> bool` and consider there is nothing more to delete if it returns `false`. That way, we save half the activities and don't have to transfer  a lot of data back to temporal.